packages/mds-event-processor/src/proc-event.ts

```diff
@@ -0,0 +1,276 @@
+import db from '@mds-core/mds-db'
+import cache from '@mds-core/mds-cache'
+import log from '@mds-core/mds-logger'
+
+import {
+  CE_TYPE,
+  InboundEvent,
+  InboundTelemetry,
+  StateEntry,
+  TripEvent,
+  TripTelemetry,
+  Telemetry,
+  EVENT_STATUS_MAP,
+  VEHICLE_EVENT,
+  VEHICLE_REASON
+} from '@mds-core/mds-types'
+import { getAnnotationData, getAnnotationVersion } from './annotation'
+import { dataHandler } from './proc'
+
+/*
+    Event processor that runs inside a Kubernetes pod.
+    Streams cloudevents from mds agency and process them in multiple ways:
+
+        1) quality checks
+        2) status changes
+        3) trip identification
+
+    Processed events/telemetry are added to various caches keyed as follows:
+
+        1) device:state (latest event/telemetry for a device)
+        2) trip:events (events linked to trips of a device)
+        3) device:ID:trips (all telemetry linked to all trips of a device)
+            - ID is the combination 'provider_id:device_id'
+
+    A Postgres table is also populated to store historical states:
+
+        REPORTS_DEVICE_STATES:
+          PRIMARY KEY = (provider_id, device_id, timestamp, type)
+          VALUES = deviceState
+*/
+
+async function getTripId(deviceState: StateEntry) {
+  /*
+  Return trip_id for telemetery entry by associating timestamps
+  */
+  const { provider_id, device_id, timestamp } = deviceState
+  const tripsEvents = await cache.readTripsEvents(`${provider_id}:${device_id}`)
```

Comment:
For my own edification, why not fall back to the `report_trips` table in the DB if there's a cache miss?